wom/src/main/scala/wdl4s/wom/graph/UnInstantiatedExpressionNode.scala

```diff
@@ -0,0 +1,30 @@
+package wdl4s.wom.graph
+
+import wdl4s.wdl.types.WdlType
+import wdl4s.wom.expression.WomExpression
+import wdl4s.wom.graph.GraphNodePort.{GraphNodeOutputPort, InputPort}
+
+
+/**
+  * Used to hold an un-instantiated expression. This is useful for a task declaration for example
+  * for which expression inputs are not explicitly mapped to other output ports but are instead derived from preceding
+  * inputs.
```

Comment:
It feels a little weird that we need this as a separate concept. Might it make sense to have a lazy evaluation that could be used in this case to yield output ports for preceding inputs? Or maybe just a name that better reflects why this concept exists and not just that it means the expression isn't instantiated.

Reply:
Yeah the reason for that is that I wanted to have a `Map[InputDefinition, OutputPort]` rather than a `Map[InputDefinition, Either[OutputPort, WomExpression]]` or 2 maps.
And to have an OutputPort one needs a GraphNode to hold it, hence this class.

But I guess we could have 2 maps or one with a complex value type and then we wouldn't need this.